db/db.go

```diff
@@ -142,12 +180,13 @@ func (db *pgDB) StoreUnresolvedBatchKeys(ctx context.Context, bks []types.BatchK
 		return err
 	}
 
+	stmt, err := tx.PreparexContext(ctx, storeUnresolvedBatchesSQL)
```

Comment:
Just curious. We can not prepare statements in the constructor for sql queries that store data? Just the ones that  retrieve data? 

I am asking this because I see you are not preparing statements for some queries.

Reply:
It is executed within the transaction so preparing it in the contractor is not doable. I'll change this logic in the next PR and all queries will be prepared in the constructor.